src/thor/route_matcher.cc

```diff
@@ -60,10 +60,9 @@ float length_comparison(const float length, const bool exact_match) {
 //
 // Get a map of end edges and the start node
 // of each edge. This is used to terminate the edge walking method.
-end_node_t GetEndEdges(GraphReader& reader,
-                       const google::protobuf::RepeatedPtrField<valhalla::Location>& correlated) {
+end_node_t GetEndEdges(GraphReader& reader, const valhalla::Location& destination) {
   end_node_t end_nodes;
-  for (const auto& edge : correlated.rbegin()->path_edges()) {
+  for (const auto& edge : destination.path_edges()) {
```

Comment:
this was just a refactory for clarity, here we are only concerned about the destination so i refactored to have the function have you send just the destination